pkg/questionModel/questionModel.go

```diff
@@ -0,0 +1,208 @@
+package questionModel
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"simple-ec2/pkg/ec2helper"
+	"strings"
+
+	"github.com/charmbracelet/bubbles/list"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/glamour"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/olekukonko/tablewriter"
+)
+
+const defaultWidth = 20
+
+var (
+	noStyle           = lipgloss.NewStyle()
+	xSmallLeftPadding = lipgloss.NewStyle().PaddingLeft(1)
+	smallLeftPadding  = lipgloss.NewStyle().PaddingLeft(3)
+	mediumLeftPadding = lipgloss.NewStyle().PaddingLeft(5)
+	largeLeftPadding  = lipgloss.NewStyle().PaddingLeft(7)
+	xLargeLeftPadding = lipgloss.NewStyle().PaddingLeft(9)
+	focused           = lipgloss.NewStyle().Foreground(lipgloss.Color("170"))
+	blurred           = lipgloss.NewStyle().Foreground(lipgloss.Color("240"))
+	paginationStyle   = list.DefaultStyles().PaginationStyle.PaddingLeft(4)
+	helpStyle         = list.DefaultStyles().HelpStyle.PaddingLeft(4).PaddingBottom(1)
+	quitTextStyle     = lipgloss.NewStyle().Margin(1, 0, 2, 4)
+	errorStyle        = lipgloss.NewStyle().Foreground(lipgloss.Color("9"))
+)
+
+type CheckInput func(*ec2helper.EC2Helper, string) bool
+
+type BubbleTeaData struct {
+	DefaultOption     string
+	DefaultOptionList []string
+	OptionData        [][]string
+	HeaderStrings     []string
+	IndexedOptions    []string
+	QuestionString    string
+	EC2Helper         *ec2helper.EC2Helper
+	Fns               []CheckInput
+}
+
+type bubbleModel interface {
+	InitializeModel(data *BubbleTeaData)
+	Init() tea.Cmd
+	Update(msg tea.Msg) (tea.Model, tea.Cmd)
+	View() string
+	getError() error
+}
+
+type item string
+
+func (i item) FilterValue() string { return "" }
+
+type itemDelegate struct {
+	renderUnselected func(str string, index int) string
+	renderSelected   func(str string, index int) string
+}
+
+func (d itemDelegate) Height() int                               { return 1 }
+func (d itemDelegate) Spacing() int                              { return 0 }
+func (d itemDelegate) Update(msg tea.Msg, m *list.Model) tea.Cmd { return nil }
+func (d itemDelegate) Render(w io.Writer, m list.Model, index int, listItem list.Item) {
+	i, ok := listItem.(item)
+	if !ok {
+		return
+	}
+
+	str := d.renderUnselected(string(i), index)
+	if index == m.Index() {
+		str = d.renderSelected(string(i), index)
+	}
+
+	fmt.Fprintf(w, str)
+}
+
+func AskQuestion(model bubbleModel, questionData *BubbleTeaData) error {
+	fmt.Println()
+	model.InitializeModel(questionData)
+	p := tea.NewProgram(model)
+	if err := p.Start(); err != nil {
+		fmt.Printf("Alas, there's been an error: %v", err)
```

Comment:
This looks like it was copied and pasted from someone else's code, which is fine, but we may need to verify that their code is under an acceptable license for us to re-use it in this project. Do you know the source?

Reply:
I found this code on the readme for bubble tea. I can change it to be less blatantly obvious